### ISSUE TITLE
Failed to generate glTF Material using glTF 2.0 add-on

### ISSUE THREAD

**Issue by themasmedia** [saturday06/VRM-Addon-for-Blender#533]
## Describe the bug

Textures and shading network fail to export with VRM export.

## To Reproduce

1. Prepare armature and mesh(es) according to VRM requirements.
2. Assign material(s) using Principled BSDF with image texture nodes connected to BaseColor, Roughness, Alpha, etc.
3. Go to 'File > Export > VRM > (.vrm)'
4. See error: ERROR:bl_ext.blender_org.vrm.exporter.vrm0_exporter:[VRM Add-on:ERROR] Failed to generate glTF Material using glTF 2.0 add-on

```
ERROR:bl_ext.blender_org.vrm.exporter.vrm0_exporter:[VRM Add-on:ERROR] Failed to generate glTF Material using glTF 2.0 add-on
Traceback (most recent call last):
  File "C:\Users\<redacted>\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\blender_org\vrm\exporter\vrm0_exporter.py", line 1275, in make_non_vrm_mat_dict
    gltf2_io_material, _ = gather_material(
                           ^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 37, in wrapper_cached
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_materials.py", line 89, in gather_material
    pbr_metallic_roughness, uvmap_info_pbr_metallic_roughness, vc_info, udim_info_prb_mr = __gather_pbr_metallic_roughness(
                                                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_materials.py", line 465, in __gather_pbr_metallic_roughness
    return gltf2_blender_gather_materials_pbr_metallic_roughness.gather_material_pbr_metallic_roughness(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 37, in wrapper_cached
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_materials_pbr_metallic_roughness.py", line 30, in gather_material_pbr_metallic_roughness
    base_color_texture, uvmap_info, udim_info_bc, _ = __gather_base_color_texture(blender_material, export_settings)
                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_materials_pbr_metallic_roughness.py", line 150, in __gather_base_color_texture
    tex, uvmap_info, udim_info, factor = gather_texture_info(inputs[0], inputs, export_settings)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_texture_info.py", line 30, in gather_texture_info
    return __gather_texture_info_helper(primary_socket, blender_shader_sockets, 'DEFAULT', filter_type, export_settings)           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 37, in wrapper_cached
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_texture_info.py", line 70, in __gather_texture_info_helper
    index, factor, udim_image = __gather_index(blender_shader_sockets, None, export_settings)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_texture_info.py", line 226, in __gather_index
    return gltf2_blender_gather_texture.gather_texture(blender_shader_sockets, use_tile, export_settings)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 37, in wrapper_cached
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_texture.py", line 45, in gather_texture
    sampler=__gather_sampler(blender_shader_sockets, export_settings),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\BlenderFoundation.Blender_4.2.0.0_x64__ppwjx1n5r4v9t\Blender\4.2\scripts\addons_core\io_scene_gltf2\blender\exp\material\gltf2_blender_gather_texture.py", line 173, in __gather_sampler
    export_settings['log'].warning(
    ~~~~~~~~~~~~~~~^^^^^^^
KeyError: 'log'
```

## Expected behavior

gLTF 2.0 textures to generate with VRM export.

## Desktop (please complete the following information)

- OS: Windows 10
- Blender 4.2
- Addon Version 2.20.70

## Additional context

Error started happening in Blender 4.,2

**Comment by saturday06** [comment]
Thank you for the report!

This has been fixed in the latest version 2.20.71 just released. Please try it.

**Comment by themasmedia** [comment]
So quick. Amazing work!
Works perfectly now.
Thank you!!! 🙏
